### PR TITLE
Interactive plotting

### PR DESCRIPTION
**Pull request by lintusj1**
```
m = ma2.get_model()
vis = dict(xlim=[-2,2], ylim=[-1,1])

r = elfi.Rejection(m, 'd')
s = r.sample(1000, threshold=.15, vis=vis)
```

### CODE REVIEW

File: elfi/methods/methods.py [hunk 1]
```diff
@@ -515,6 +544,16 @@ def _update_objective(self):
         self.objective['n_batches'] = ceil(n_batches)
         logger.debug('Estimated objective n_batches=%d' % self.objective['n_batches'])
 
+    def plot_state(self, **options):
+        displays = []
+        if options.get('interactive'):
+            from IPython import display
```
Comment (vuolleko) [inline elfi/methods/methods.py]: Maybe put the import in the beginning of `infer` where it checks if `vis`was given.

File: elfi/model/elfi_model.py [hunk 2]
```diff
@@ -247,14 +247,23 @@ def __init__(self, *parents, state=None, model=None, name=None):
...
+            If name ends in an asterisk '*' character, the asterisk will be replaced with
+            a random string and the name is ensured to be unique within the model.
         state : dict
         model : ElfiModel
+
+        Examples
+        --------
+        >>> node = NodeReference(name='name*')
+        >>> node.name
+        name_12fr3s
```
Comment (vuolleko) [inline elfi/model/elfi_model.py]: Maybe unnecessary to doctest random stuff (especially this fails now)?

File: elfi/plotting.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+def get_axes(**options):
+    if 'axes' in options:
+        return options['axes']
+    return plt.gca()
+
+
+def _update_interactive(displays, options):
+    displays = displays or []
+    if options.get('interactive'):
+        from IPython import display
```
Comment (vuolleko) [inline elfi/plotting.py]: If this file is only for interactive plotting, importing can be at the top.

Comment (lintusj1) [inline elfi/plotting.py]: I was planning this to be for plotting in general.

File: elfi/methods/methods.py (hunk 1 above)
Comment (lintusj1) [inline elfi/methods/methods.py]: You can call this method also directly if you loop the inference yourself, or just want the IPython plotted version so the import should be here.

**Pull request merged**